samples/Samples.Mvc5/Controllers/HomeController.cs

```diff
@@ -131,6 +133,34 @@ public ActionResult About()
         /// <returns>The <see cref="ActionResult"/>.</returns>
         public ActionResult ResultsAuthorization() => View();
 
+        public async Task<ActionResult> CallAsync()
+        {
+            var profiler = MiniProfiler.Current;
+            var service = new FooService();
+            using (profiler.Step("action"))
+            {
+                var head = profiler.Head;
+                var tasks = new List<Task>();
+                for (int i = 0; i < 10; i++)
+                {
+                    tasks.Add(service.FooAsync(head));
+                }
+                await Task.WhenAll(tasks);
+                return Content("All good");
+            }
+        }
+
+        class FooService
+        {
+            public async Task FooAsync(Timing head)
+            {
+                using (new Timing(MiniProfiler.Current, head, "foo async", minSaveMs: 100000))
```

Comment:
if i use profiler.Step for this it would make all foo async timing to be a long tree with each timing having one child only, this is because Timing replaced the head with itself, so creating 1000 timings in parallel would make them all children of each other. My test is a bit more artificial than i would have liked - i explicitly pass the correct head to each timing